src/utils/actionUtils.ts

```diff
@@ -99,7 +99,7 @@ export function getInputS3ClientConfig(): S3ClientConfig | undefined {
                 process.env["AWS_SESSION_TOKEN"]
         },
         region: core.getInput(Inputs.AWSRegion) || process.env["AWS_REGION"],
-        endpoint: core.getInput(Inputs.AWSEndpoint),
+        endpoint: core.getInput(Inputs.AWSEndpoint) || undefined,
```

Comment:
We should change here because empty string throws error `Invalid URL`. 
https://github.com/smithy-lang/smithy-typescript/blob/d952a6a9d6a436b79f6174fcee9e4fbed82ca841/packages/middleware-endpoint/src/resolveEndpointConfig.ts#L103-L122